### DISCUSSION TITLE
[SOLVED] RSS Guard supports plugins, can scrape websites with scripts

### DISCUSSION THREAD

**Discussion by martinrotter** [martinrotter/rssguard#578] [General]
@Owyn

Saw your ticket on quiterss repo. I would like to inform you that RSS Guard does support those features you requested for quiterss. Documentation:

https://github.com/martinrotter/rssguard/blob/master/resources/docs/Documentation.md#scrap
https://github.com/martinrotter/rssguard/blob/master/resources/docs/Documentation.md#fltr

**Comment by Owyn** [comment]
It doesn't  seem to support any accessibility at all, look at this forced super-small unchangable text size, it even ignores windows zoom and text size settings (you can see that with menu text size which was left alone in comparison with enforced small text size in the feed categories list and app settings):
![image](https://user-images.githubusercontent.com/1309656/148936655-2b571fb3-4e41-486e-9249-c76c05804ef6.png)

I looked through all settings but there was nothing about text size or scaling

edit: searching through issues helped find the magic setting but not much, settings font is still unchangable: 
![image](https://user-images.githubusercontent.com/1309656/148937481-beca40aa-fd91-47de-b3a8-98a6bee03256.png)

**Reply by martinrotter** [reply]
You want to be able to sent FONT SIZE application-wide? For all texts in app? Also, what is "windows zoom"?

**Reply by Owyn** [reply]
yea, that's how it usually is in most applications, a setting in "Interface" for that

windows zoom level:
![image](https://user-images.githubusercontent.com/1309656/148941237-4f714045-dcd1-4f84-a656-5ecc85483328.png)

**Comment by Owyn** [comment]
Also internal web-browser seems to have hardcoded colors and ignore windows (contrast) theme colors entirely:
![image](https://user-images.githubusercontent.com/1309656/148941059-dfd308b5-4b42-4d2f-826a-b71ce8948a32.png)

**Comment by martinrotter** [comment]
1. Internal web browser uses colors from active "Skin" which is changeable in settings. There is dark skin (dark, or nudus-dark) and light skin available.
2. You can zoom internal web browser and zoom state is remembered.
3. You can change font size globally by using environment variable `QT_SCALE_FACTOR`. Set it either globally in your PC, or locally for RSS Guard for example to value `2`. Result likes like pic below. Here is some related [documentation](https://doc.qt.io/qt-5/highdpi.html#high-dpi-support-in-qt).

![image](https://user-images.githubusercontent.com/1255302/148941474-637ebe7f-e611-4822-b34c-8ea7a6d67af6.png)

**Reply by martinrotter** [reply]
4. To follow Windows zoom settings [automatically](https://stackoverflow.com/questions/47587874/qt-auto-screen-scale-factor-has-no-effect), related [docs](https://www.qt.io/blog/2016/01/26/high-dpi-support-in-qt-5-6).

**Reply by Owyn** [reply]
> Internal web browser uses colors from active "Skin" which is changeable in settings. 

no it doesn't (I restarted):
![image](https://user-images.githubusercontent.com/1309656/148942951-28db2685-6077-45a1-a97f-27b152316833.png)

**Reply by martinrotter** [reply]
Yes, it does, set it for example to "nudus-dark" and restart.

**Reply by martinrotter** [reply]
![image](https://user-images.githubusercontent.com/1255302/148943183-bf51c7a7-cba7-4395-a8d5-f15678d55fe4.png)
![image](https://user-images.githubusercontent.com/1255302/148943210-b71ab6d3-0d0c-4200-8a96-1666223c4cc8.png)

**Reply by Owyn** [reply]
Indeed, seems like not all dark themes are actually dark 🤔

**Comment by Owyn** [comment]
When i imported all my old feeds (without fetching metadata cuz else it wouldn't have added feeds which are currently offline), now none of my feeds have a favicon unless I click "fetch metadata" on each and every feed one by one? there is no "fetch metadata for all" button??
![image](https://user-images.githubusercontent.com/1309656/148942554-19e8c2ff-be4d-459a-be20-b4f1b52d40a4.png)

**Reply by martinrotter** [reply]
No, but when you import OPML, choose "YES" to fetch metadata, it will add even those feeds which are currently online.

If you want to have feature "Fetch metadata for all feeds" added, post separate ticket.

**Reply by Owyn** [reply]
If I choose YES then it simply won't add half of my feeds cuz they are offline currently

**Reply by martinrotter** [reply]
I see, that looks like a bug. Can you provide your OPML? Or create some minimal OPML which shows the bug with some offline feeds perhaps pls?

**Reply by Owyn** [reply]
[f.zip](https://github.com/martinrotter/rssguard/files/7846546/f.zip)
my opml files with some offline feeds

it seems intentional tho, cuz it says "some feeds weren't added ..., see debug log"

**Reply by martinrotter** [reply]
let me check

**Reply by martinrotter** [reply]
here it imports eevrything with your opml file, even offline feeds, as it should

![image](https://user-images.githubusercontent.com/1255302/148946516-10abfe0b-4f2b-49ec-84b2-ab1a90a9183e.png)
 i get no error, can you provide debug log?

**Reply by martinrotter** [reply]
https://github.com/martinrotter/rssguard/blob/master/resources/docs/Documentation.md#reprt

**Reply by Owyn** [reply]
hmmm wait now it seems to work but first time it  definately said something about "see debug log, some feeds didn't load"

**Reply by Owyn** [reply]
Not all favicons seemed to load, even when I press "load metadata" specifically after
examples: kirei cake, kuma translations, dead mount

Looking at your screen - you seem to have the same problem with loading favicons

**Reply by Owyn** [reply]
I tried to make a log file like docs say :
![image](https://user-images.githubusercontent.com/1309656/148957970-6a1a25c1-8bfe-43f1-81d5-586296cae825.png)
but the log file simply wasn't created anywhere after the program ran (write rights on the folder are free)
![image](https://user-images.githubusercontent.com/1309656/148958271-ceb721cf-a6f8-441e-8c08-0a6bf15ba57d.png)

**Reply by martinrotter** [reply]
that sounds weird, double check for the log, should be there

**Reply by martinrotter** [reply]
also, "Kumo desu ga" is the, just with fetched title "Nyaa....."

as for favicons, let me check

**Reply by Owyn** [reply]
That also seems weird, why did it reset my custom feed name by itself?

**Reply by martinrotter** [reply]
> That also seems weird, why did it reset my custom feed name by itself?

Probably because metadata were fetched during import?

**Comment by Owyn** [comment]
![image](https://user-images.githubusercontent.com/1309656/148944251-247d833d-8f30-4873-9128-749725ad190a.png)

how do I make article text in center not at right? I don't need read\importance stuff, how do I disable that panel?

**Reply by martinrotter** [reply]
![image](https://user-images.githubusercontent.com/1255302/148955590-619616e0-3786-4e4d-abc0-8504150290e8.png)

**Reply by martinrotter** [reply]
>how do I make article text in center not at right

Article text is formatted with skin CSS, almost all skins format texts "justified" -> not aligned to right, ellaborate what exactly you mean.

**Reply by Owyn** [reply]
That was it, now text is in the center (of the monitor)
I was expecting I could do that by right clicking the panel and choosing "hide" as most apps do

**Reply by martinrotter** [reply]
which most apps, can you name them?

**Reply by Owyn** [reply]
sorry, it's just that last such toolbar I remember was hidden that way iirc, 
but yea, view menu -> show\hide is also the usual way

**Comment by Owyn** [comment]
![image](https://user-images.githubusercontent.com/1309656/148956623-28ffe8fd-54f1-43b1-b7a8-c41ce7b35126.png)

My colors look pretty colorful for a dark theme, the blue color especially seems rather unreadable

**Reply by martinrotter** [reply]
try to switch rather to "nudus-dark" as dark theme is VERY old and not maintained anymore, it will be removed very soon, switch to nudus-dark and report back if any problems

**Reply by Owyn** [reply]
That is better, thought - half of it uses system contrast colors while other half (browser) - doesn't
![image](https://user-images.githubusercontent.com/1309656/148960758-6666d7c9-12b3-447c-a66b-a6a684d1bcf3.png)
p.s. - why is the [x] unavailable and only Ok\Cancel can be used to close the settings window? 😆

**Reply by martinrotter** [reply]
because that is the UX approach all modal dialogs use, you can either OK or CANCEL, you can use ESC button.

**Reply by Owyn** [reply]
but why disable the [x]? ppl love pressing [x]... 😢

**Reply by martinrotter** [reply]
if more people will show interest in X buttons I will implement, so far, you are the first one requesting this

**Reply by Owyn** [reply]
my parents for example never press ok\cancel, they always press [x] 😄

**Reply by martinrotter** [reply]
Yes, they won't be typical RSS Guard users I guess.

**Comment by Owyn** [comment]
> @Owyn
> 
> Saw your ticket on quiterss repo. I would like to inform you that RSS Guard does support those features you requested for quiterss. Documentation:
> 
> https://github.com/martinrotter/rssguard/blob/master/resources/docs/Documentation.md#scrap https://github.com/martinrotter/rssguard/blob/master/resources/docs/Documentation.md#fltr

I tried one of the examples `php#tweeper.php#-v#0#https://twitter.com/NSACareers` and it didn't work:
![image](https://user-images.githubusercontent.com/1309656/148970228-8b5f3a24-7880-43d2-aa5f-1995e27d6ed4.png)

I wish it was something simple here to parse websites into RSS, like feed-me-up-scotty https://feed-me-up-scotty.vincenttunru.com/ (maybe you could incorporate that script into RSSguard? the source is on the gitlab) , there I just had to specify a CSS selector where all the needed `<a>` links are and it made those links into feed articles:
````
[EvilGod]
title = "I'm an Evil God chapters"
url = "https://mangahub.io/manga/i-m-an-evil-god/"
entrySelector = "ul.list-group li"
titleSelector = "a span"
linkSelector = "a"
contentSelector = "a span"
````

Output: https://github.com/Owyn/feeds/blob/gh-pages/EvilGod.xml

how do I do the same here? examples seem to be very lacking for practical use...

**Reply by martinrotter** [reply]
You need to have Tweeper and Php installed for that command to work.

As for Feed-me-up script, it is Node.js based and can be run through "web scrape" feature of RSS Guard now to generate almost real-time results, no changes needed. But as said in documentation - IT IS for advanced users only. Why? Because there is no universal solution to scrape websites. Read documentation, there are even sample scripts placed in repository (linked from documentation).

**Reply by martinrotter** [reply]
I will make and example for you for feed-me-scotty.

### Setup Node.JS (this is independent of RSS Guard)
1. Install Node.JS and NPM (included in installer).
2. Run command `npm list -g --depth=0` in command line, it spits out global path for NPM packages.
3. Add the path from previous step to `PATH` environment variable.
4. Run command `npm root -g` and use its path as value for `NODE_PATH` variable.
![image](https://user-images.githubusercontent.com/1255302/149081429-4573026d-4fb5-4491-90d5-2fb50cd6c8f1.png)Open command line and install `feed-me-up-scotty` with `npm -g i feed-me-up-scotty`.
6. Now, go to some directory, for example `C:\feeds\` and create `feeds.toml` file with some settings there.
7. Try to run command `npx feed-me-up-scotty 'c:\feeds\feeds.toml'`. It should run for a few seconds, then write `Feeds generated in `public/`.` output.
8. Your XML files are now generated in `public` subfolder and your setup is complete, now to RSS Guard.

### Setup `feed-me-up-scotty` to use with RSS Guard.
1. Now, we need to tell RSS Guard to run that NPX `feed-me-up-scotty` and read resulting XML.
2. Create script in PowerShell `run-scotty.ps1` and paste this contents in it, tweak it to suit your needs:
```powershell
$base = "c:\feeds"

Set-Location "$base"
Remove-Item -Recurse -Force -Confirm:$false "$base\public" > $null 2> $null

npx feed-me-up-scotty "$base\feeds.toml" > $null 2> $null

Get-Content "$base\public\all.xml"
```
3. Test calling the script in PowerShell CLI.
4. Open RSS Guard, `Add new feed` and setup it like this, adjust paths to place where you have your script:
![image](https://user-images.githubusercontent.com/1255302/149081457-4060e555-0192-48f3-bb2a-3b810931b53a.png)
5. Run `Fetch metadata`, click `OK`.
6. Tweak to your needs. Script gets called automagically each time feed is fetched in RSS Guard and feeds are regenerated each time with new data.
![image](https://user-images.githubusercontent.com/1255302/149081650-d685e9b6-88c1-44c4-9241-c604bdcf3f1e.png)

**Reply by Owyn** [reply]
Well it all worked fine till the last step... 
![image](https://user-images.githubusercontent.com/1309656/149102289-d9f6b617-1841-415c-87b9-6b594eb4017c.png)
and then:
![image](https://user-images.githubusercontent.com/1309656/149104986-29a3b01e-581e-4dc5-af82-eb7b3fc5ffe7.png)

even tho xml file all.xml looks fine 
````
<?xml version="1.0" encoding="utf-8"?>
<feed xmlns="http://www.w3.org/2005/Atom">
    <id>https://example.com/all.xml</id>
    <title>Combined feed</title>
    <updated>2022-01-12T09:26:25.308Z</updated>
    <generator>https://github.com/jpmonette/feed</generator>
    <entry>
        <title type="html"><![CDATA[Chapter 221.2 (I'm an Evil God)]]></title>
        <id>https://manhuaus.com/manga/im-an-evil-god/chapter-221-2/</id>
        <link href="https://manhuaus.com/manga/im-an-evil-god/chapter-221-2/"/>
        <updated>2022-01-12T09:26:24.638Z</updated>
        <content type="html"><![CDATA[Chapter 221.2]]></content>
    </entry>
</feed>
````

and then when I try again it again says script takes too long

**Reply by Owyn** [reply]
> But as said in documentation - IT IS for advanced users only. Why? Because there is no universal solution to scrape websites. Read documentation, there are even sample scripts placed in repository (linked from documentation).

using CSS selectors **is** a rather universal solutions to scrape websites tho... pretty much every service (website to RSS) uses it when you google around how to make an RSS out of a usual website and find a website-service offering you a way

I wish there was a built-in thing in RSSguard for that exact purpose to make things easy & fast so we just had to input CSS selectors and it all worked right away - that way it would no longer be only for super-advanced users

and even non-advanced users can understand full examples and simple instructions, like the one you posted above

**Reply by martinrotter** [reply]
> 

Does the powershell script work and exit correctly when you run it directly in powershell? also, try to hit "Fetch metadata" button several times, the script sometimes really takes long, this is probably some bug in feed-me-scotty program. After several attempts, it worked.

**Reply by martinrotter** [reply]
ALSO: Marking this as solved. PLEASE, create SEPARATE tickets/discussions for each problem/suggestion you have.

Do not COUPLE several topics into single ticket/discussion, it makes things messy and easy to overlook.

**Reply by martinrotter** [reply]
> > But as said in documentation - IT IS for advanced users only. Why? Because there is no universal solution to scrape websites. Read documentation, there are even sample scripts placed in repository (linked from documentation).
> 
> using CSS selectors **is** a rather universal solutions to scrape websites tho... pretty much every service (website to RSS) uses it when you google around how to make an RSS out of a usual website and find a website-service offering you a way
> 
> I wish there was a built-in thing in RSSguard for that exact purpose to make things easy & fast so we just had to input CSS selectors and it all worked right away - that way it would no longer be only for super-advanced users
> 
> and even non-advanced users can understand full examples and simple instructions, like the one you posted above

If you have this solution with feed-me-scotty almost ready, you can easily generalize it to use it with multiple websites easily, I will not "bundle" this to RSS GUard directly because that would add yet another bunch of nasty Node.js dependencies which I think about several times before including them into project.

**Reply by Owyn** [reply]
> > 
> 
> Does the powershell script work and exit correctly when you run it directly in powershell? also, try to hit "Fetch metadata" button several times, the script sometimes really takes long, this is probably some bug in feed-me-scotty program. After several attempts, it worked.

yea it does, works fine and creates files fine, even when run from a powershell or from the rssguard when it gives an error (xml files are still created fine)

what If I want to add several feeds into several categories in the rssguard tho? would I need to run the script and feed-me-up several times in the rssguard?

> SEPARATE tickets/discussions for each problem/suggestion you have.

this is just the discussion thread of me trying to initially set up RSSGuard 😆

**Reply by Owyn** [reply]
> another bunch of nasty Node.js dependencies

that's totally understandable, that's why I suggesting to prepackage "something similar", just the bare mimimum to use CSS selectors to crawl websites without an RSS feed available, which would work fast and easily 

- it'd be very user-friendly and useful which'd finally make this feature of scraping websites go live 1.0 for public

**Reply by martinrotter** [reply]
> yea it does, works fine and creates files fine, even when run from a powershell or from the rssguard when it gives an error (xml files are still created fine)

Just testing too. It seems to me that somehow, the script is much more unrealiable when executed "externally". It is feed-me-scotty which just hangs (waiting for website to load in its headless browser). Try repeatedly.

**Reply by martinrotter** [reply]
> that's totally understandable, that's why I suggesting to prepackage "something similar", just the bare mimimum to use CSS selectors to crawl websites without an RSS feed available, which would work fast and easily

RSS Guard already uses Node.JS components for Adblocking functionality. There will be some updates to how node.js is integrated within the app, after that gets implemented, I might consider adding this, but it is highly unlikely, maybe I will implement some very bare feature like this by myself.

**Reply by Owyn** [reply]
It works fine for github and gitlab workers but for home computers - no idea, might not be as good,

I thought considering you authored the whole RSSGuard and know it well - you'd know a fast & reliable way or a tool to scrape websites from within it

**Reply by Owyn** [reply]
> RSS Guard already uses Node.JS components for Adblocking functionality.

I don't need that btw... I hope there is \ would be a way to disable that (I disabled everything in web-engine settings already), along with the whole browser thing - I need just the text RSS feeds give me and `<img>` images sometimes, not the whole browser... I think quiterss supports disabling internal browser entirely and showing just news description

**Reply by martinrotter** [reply]
> It works fine for github and gitlab workers but for home computers - no idea, might not be as good,
> 
> I thought considering you authored the whole RSSGuard and know it well - you'd know a fast & reliable way or a tool to scrape websites from within it

THat feed-my-scotty seems to be highly unrealiable. It depends whether the website USES Javascript to load the content or not. If not, then you will simply be good with Python and https://beautiful-soup-4.readthedocs.io/en/latest/

If site uses Javascript, then you need some headless browser to first fetch the HTML DOM fully loaded and for that and Python, https://github.com/pyppeteer/pyppeteer is probably one of the best solutions.

If you want to go Node.js way, then there is pUppeteer (not pYppeteer) or selenium.

There are many headless browsers.

**Reply by martinrotter** [reply]
> 

It is not enabled by default, don't worry.

You can also use no-webengine version.

**Reply by Owyn** [reply]
How do I use the no-webengine version?

**Reply by martinrotter** [reply]
https://github.com/martinrotter/rssguard/blob/master/resources/docs/Documentation.md#webb

**Reply by Owyn** [reply]
found it in releases! thank you

but can it at least display images? I don't need much, just images cuz some feeds are simple a feed of one image per item where artists post their new works

In documentation example image -  images seem to be gone

**Reply by Owyn** [reply]
> If not, then you will simply be good with Python

Most websites work fine without the JS as far as I've experienced, everything needed usually is in the page source,

As I've checked crawler scripts in python ( https://github.com/martinrotter/rssguard/tree/master/resources/scripts/scrapers ) - and all of  them seem to be in raw python without any soup? 

why not just use those good examples you've posted - beautiful soup \ pyppeteer ?

soup itself even seems to have a built-in support for CSS selectors it seems https://stackoverflow.com/questions/11465555/can-we-use-xpath-with-beautifulsoup

**Reply by martinrotter** [reply]
> found it in releases! thank you
> 
> but can it at least display images? I don't need much, just images cuz some feeds are simple a feed of one image per item where artists post their new works
> 
> In documentation example image - images seem to be gone

No, at this point it only show clickable links which you can display in external app/browser. If you want a way to show images, post ticket pls.

**Reply by martinrotter** [reply]
> As I've checked crawler scripts in python ( https://github.com/martinrotter/rssguard/tree/master/resources/scripts/scrapers ) - and all of them seem to be in raw python without any soup?

Some of those are using https://newspaper.readthedocs.io/en/latest/ and some of them are using BeautifulSoup. It is really just a bunch of scripts to be used as inspiration.

**Reply by Owyn** [reply]
> It is really just a bunch of scripts to be used as inspiration.

I mean, the soup can be presented as a default to-go way to do things easily here if it's good

**Reply by martinrotter** [reply]
bsoup works nicely, you should used that if sites do not use JS

**Comment by Owyn** [comment]
about the main theme of the topic again - 

> plugins, can scrape websites with scripts 

but can it use compiled executables instead of scripts? cuz scripts can be compiled, or just some non-scripting language compiled. and also compiled scripts'd be faster than non-compiled ones, and users wouldn't need to install any development environment like scripting languages onto their PC

**Reply by martinrotter** [reply]
If course it can. It can call just about any valid executable.

**Reply by Owyn** [reply]
That would be a great way for the users

**Reply by martinrotter** [reply]
As I said before, you can call just anything.
![image](https://user-images.githubusercontent.com/1255302/149500979-8ca8479b-f023-43b2-b750-57fae51cc619.png)

**Comment by Owyn** [comment]
> The # character is there to separate interpreter and individual arguments. I had to select some character as separator because simply using space is not that easy as it might sound, as sometimes space could be a part of an argument itself.

That's a problem, since # sign is used for Id CSS selectors 😢  so I can't pass CSS selectors as arguments now when I'm pretty much done with my python script analogue of feed-me-up

why not just us `"` and `'` just like how all passing of parameters are done everywhere?


and 2nd) - does the json format RSSguard uses for feed supports baseUrl? if links \ favicons are relative (especially favicons, cuz most has relative links)

if it's a post-proccess script and it gets no url at all, so RSSguard should provide the baseUrl itself then... cuz now relative links in json don't work

**Comment by Owyn** [comment]
Here, I made this for RSSGuard: https://github.com/Owyn/CSS2RSS

**Reply by martinrotter** [reply]
Nice, so you solved the "#" problem somehow? It is used for "ID" attribute right?

**Reply by martinrotter** [reply]
The script is nice. I will make the textbox for scripts bit bigger.

**Reply by martinrotter** [reply]
a78d2852d

**Reply by Owyn** [reply]
> Nice, so you solved the "#" problem somehow? It is used for "ID" attribute right?

I didn't :D I thought you'd just change the symbol (or make a setting to choose the symbol itself) after hearing my thoughts here... or make it use standard quotas `'` & `"` for arguments with spaces

Thought I could just use some other symbol like `@` for CSS tags, and script would replace `@` in arguments to `#` but that would be weird and unhandy for users, e.g. - they click "copy CSS selector for this element" and just enter it into arguments and then wonder why it doesn't work cuz they forgot about this workarounding...

**Reply by martinrotter** [reply]
Not so easy to use `'` or `"` because they can be used inside arguments as well.

Your workaround is rather hacky way. I would like to solve this efficiently. Perhaps, you could "URL-encode" those `#` symbols? In encoded form, they are `%23` which is de-facto standard for "encoding" in our environment. Script can then easily decode.

I need to keep backwards compatibility to not break things. You are first user who mentions problems with `#`. Perhaps I could make it changeable in settings. If you feel it would work, create separate ticket for it please. :)

**Reply by Owyn** [reply]
> Not so easy to use `'` or `"` because they can be used inside arguments as well.

yea, but not together, also there's escaping, most apps & systems use quotas for arguments with spaces, so why the innovative way with the #?

> Script can then easily decode.

It's the same hacky thing for users then, they'd need to remember to replace already ready-to-use css selectors in a hacky way replacing `#` with `%23` or smth else

**Reply by martinrotter** [reply]
Yes, post ticket, I will make it changeable.

Note, that the underlying API I use, needs to have arguments "separated", I cannot know where exactly one argument ends and another begins - yes, in most cases they are separated by "space", and quoted if there is a space inside an argument - but that approach has flaws too.

**Reply by martinrotter** [reply]
EDIT: I FOUND A WAY. Will you test it after I implement?

**Reply by martinrotter** [reply]
@Owyn OK I pushed a fix - 66159f42a.

Now, you don't have to use separators at ALL. If some of your arguments contain spaces, then enclose it with double quotes (NOT single quotes).

Pls, test and let me know, it is compilling RN.

![image](https://user-images.githubusercontent.com/1255302/149778090-6f01f97d-298e-4d50-a459-06cb4ff2753d.png)

**Reply by Owyn** [reply]
Just a question... where should I look for compiled dev builds?

**Reply by martinrotter** [reply]
https://github.com/martinrotter/rssguard/releases/tag/devbuild

right now they are not there, it is compilling RN, wait about 15 minutes, they appear as files there

**Reply by Owyn** [reply]
Pressed "what are scripts"? then pressed it again, - the window is still out of bounds, and since the [x] is disabled you can't even close it with the mouse now without dragging it

![image](https://user-images.githubusercontent.com/1309656/149789711-e674d943-8693-4592-9ae6-2d4e2d5db2a0.png)


but new param passing using quotes works
![image](https://user-images.githubusercontent.com/1309656/149790702-36e80838-69b7-4cc1-aafc-d0a7b8f83344.png)
